src/Illuminate/Database/Schema/Blueprint.php

```diff
@@ -180,17 +194,20 @@ protected function commandsNamed(array $names)
      */
     protected function addImpliedCommands(Connection $connection, Grammar $grammar)
     {
-        if (count($this->getAddedColumns()) > 0 && ! $this->creating()) {
-            array_unshift($this->commands, $this->createCommand('add'));
-        }
-
-        if (count($this->getChangedColumns()) > 0 && ! $this->creating()) {
-            array_unshift($this->commands, $this->createCommand('change'));
-        }
-
         $this->addFluentIndexes($connection, $grammar);
 
         $this->addFluentCommands($connection, $grammar);
+
+        if (! $this->creating()) {
+            $this->commands = array_map(
+                fn ($command) => $command instanceof ColumnDefinition
+                    ? $this->createCommand($command->change ? 'change' : 'add', ['column' => $command])
+                    : $command,
+                $this->commands
+            );
+
+            $this->addAlterCommands($connection, $grammar);
+        }
```

Comment:
Step 3. We walk through the commands and convert `ColumnDefinition`s into eighter an "add" or a "change" command.

Then we call `addAlterCommands` method.